therapy_app/models.py

```diff
@@ -0,0 +1,212 @@
+        user = self.create_user(
+            email,
+            password=password,
+        )
+        user.staff = True
+        user.admin = True
+        user.type = kwargs.get('type', 'Client')
+        user.save(using=self._db)
+        return user
+
+
+class User(AbstractBaseUser):
+    email = models.EmailField(
+        verbose_name='email address',
+        max_length=255,
+        unique=True,
+    )
+    active = models.BooleanField(default=True)
+    staff = models.BooleanField(default=False)  # a admin user; non super-user
+    admin = models.BooleanField(default=False)  # a superuser
+
+    type = models.CharField(max_length=255, choices=(
+        ('Client', 'Client'),
+        ('Therapist', 'Therapist'),
+    ))
+    journal = models.CharField(max_length=255, null=True, blank=True)   # Only for Clients
+
+    created_at = models.DateTimeField(auto_now_add=True, editable=False)
+    updated_at = models.DateTimeField(auto_now=True)
+    is_deleted = models.BooleanField(default=False)
+
+    USERNAME_FIELD = 'email'
+    REQUIRED_FIELDS = ['type']    # Email & Password are required by default.
+
+    objects = UserManager()
+
+    def get_full_name(self):
+        # The user is identified by their email address
+        return self.email
+
+    def get_short_name(self):
+        # The user is identified by their email address
+        return self.email
+
+    def __str__(self):
+        return self.email
+
+    def has_perm(self, perm, obj=None):
+        """Does the user have a specific permission?"""
+        return True
+
+    def has_module_perms(self, app_label):
+        """Does the user have permissions to view the app `app_label`?"""
+        return True
+
+    @property
+    def is_staff(self):
+        """Is the user a member of staff?"""
+        return self.staff
+
+    @property
+    def is_admin(self):
+        """Is the user a admin member?"""
+        return self.admin
+
+    @property
+    def is_active(self):
+        """Is the user active?"""
+        return self.active
+
+
+class Mapping(models.Model):
+    """
+    This model will represent the relations between client and therapist
+    if a client has mapped a therapist then this model will show an entry.
+    along with the journal access.
+    """
+    client = models.ForeignKey(User, on_delete=models.CASCADE, related_name='mapping_client',
+                               limit_choices_to={"type": "Client"})
+    therapist = models.ForeignKey(User, on_delete=models.CASCADE, related_name='mapping_therapist',
+                                  limit_choices_to={"type": "Therapist"})
+    journal_access = models.BooleanField(default=False)
+    journal_requested = models.CharField(max_length=255, choices=(
+        ("Approved", "Approved"),
+        ("Declined", "Declined"),
+        ("Pending", "Pending"),
+        ("Not Requested", "Not Requested"),
+    ), default='Not Requested')
+    created_at = models.DateTimeField(auto_now_add=True, editable=False)
```

Comment:
Not required as a change, but you can make a separate abstract model class containing the fields 'created_at', 'updated_at' and 'is_deleted', that ways you won't have to repeat them